nwbmatic/allends.py

```diff
@@ -0,0 +1,411 @@
+"""
+Class and functions for loading data from Allen Brain Atlas via the Allen Software Development Kit (allenSDK) API
+Currently only supports "Visual Coding - Neuropixels" database.
+
+The Visual Coding - Neuropixels project uses high-density extracellular electrophysiology probes to record spikes from
+a wide variety of regions in the mouse brain. Experiments were designed to study the activity of the visual cortex
+and thalamus in the context of passive visual stimuluation, but these data can be used to address a wide variety of topics.
+
+To see more info about dataset, see: https://allensdk.readthedocs.io/en/latest/visual_coding_neuropixels.html
+
+@author: Selen Calgin
+Date: 06/19/2023
+Last edit: 07/06/2023 by Selen Calgin
+
+"""
+import json
+import os
+import tkinter as tk
+from tkinter import ttk
+
+import pynapple as nap
+from allensdk.brain_observatory.ecephys.ecephys_project_cache import (
```

Comment:
Could you update the `pyproject.toml` so that it takes into consideration the packages to install for the allensdk.
The best would be to add an optional dependency part where nwbmatic + allen could be installed like this 
```bash
pip install nwbmatic[allen]`
```

Reply:
Yes can definitely do that.